autotest.pm

```diff
@@ -104,6 +104,7 @@ sub loadtest {
     $test             = $name->new($category);
     $test->{script}   = $script;
     $test->{fullname} = $fullname;
+    $test->{serial_failures} = \%::SERIAL_FAILURES // {};
```

Comment:
that's a bit strange API - I would prefer if this was set explicitly through the distribution object.

Reply:
@coolo 
Not sure if this is what you mean:

> $test->{serial_failures} = $testapi::distri->{serial_failures} // {};

Reply:
I mean instead of that global variable a specific call to $distri - how you handle it internally is not my problem.

Reply:
@coolo Done.

Created a function in the DIstribution.pm:

> $testapi::distri->set_serial_failures(soft=>['gcc version 4.8.5 (SUSE Linux)'], hard=>[qr/No iBFT detected/]);